go/vt/vtgate/planbuilder/dml_planner.go

```diff
@@ -42,10 +43,27 @@ func rewriteRoutedTables(stmt sqlparser.Statement, vschema plancontext.VSchema)
 			return false, vterrors.VT09014()
 		}
 
+		// For reference tables, managed by Vitess (Source not nil), we need to use the source table as the target.
+		if vschemaTable.Type == vindexes.TypeReference && vschemaTable.Source != nil {
+			vschemaTable, vindexTbl, _, _, _, err = vschema.FindTableOrVindex(vschemaTable.Source.TableName)
+			if err != nil {
+				return false, err
+			}
+			if vindexTbl != nil {
+				// vindex cannot be present in a dml statement.
+				return false, vterrors.VT09014()
+			}
```

Comment:
not sure if this will happen? do you have a test case for this?

Reply:
You mean the nil vindexTbl check? The same check is made above in the same function. So I assumed it was required. I can remove it, but is there any problem adding the extra check?

Reply:
Merging as is for now: assuming an additional check is safe. Can remove later on if required.